### DISCUSSION TITLE
Henry Law

### DISCUSSION THREAD

**Discussion by Santhosh91000** [GeoChemFoam/GeoChemFoam#25] [Q&A] [answered]
Hello everyone,

I hope you're all doing well.

I have a question regarding how the Henry law is applied in solvers such as interTransferFoam or interReactiveTransferFoam. Specifically, these solvers use the "Henry Law" to model the jump in concentration. As defined in the Geochemfoam paper, for two phases (1 and 2) and one dissolved species, denoted as A, we have the following relationship:

C1,A = HA * C2,A

In my particular case, phase 1 represents CO2, phase 2 is water, and A is the dissolved CO2. I'm attempting to find HA by the following expression:

HA = Solubility of CO2 in water / Initial concentration of CO2 in water

Is this approach correct? Because if it is, I'm obtaining a Henry coefficient of 7e*10^7, which appears to be quite high. (I'm working in microchannels, so perhaps this value isn't unreasonable.)

I'm bringing up this question here rather than in a chemistry forum because the definition of the Henry Law in this context seems to differ from the one in chemistry, where it's typically associated with the amount of dissolved species being proportional to the partial pressure of a gas. In this solver, it seems more like a law for modeling concentration jumps, which can work in both liquid-liquid and liquid-gas scenarios, as opposed to the conventional chemical definition that applies to gas-liquid interactions.

Please let me know if I've correctly understood this concept and whether a high Henry coefficient value in the context of interReactiveTransferFoam is something valid.

Thank you for your time.
Sincerely,
Santhosh

**Comment by julienmaes** [comment, marked as answer]
Hi Santhosh,

In GCFOAM it's C2,A=HA*C1,A (check the code or this paper https://link.springer.com/article/10.1007/s11242-021-01661-8, also on arxiv)

 in our definition of henry's constant, c2,A is the concentration of component A in kmol/m3 in the non-aqueous phase. So for pure CO2 gas at density 2 kg/m3 it would be ~2/44~0.045 kmol/m3. So HA= concentration of CO2 in non-aqueous phase/solubility of CO2 in water ~ 0.045 kmol/m3 / 0.036 kmol/m3 ~ 1.25. Of course this will depends on pressure, temperature, etc

**Reply by Santhosh91000** [reply]
Hello Julien,

Thanks for the response! I get it now.
I have just a subsidiary question. I was wondering if the units used in GeochemFoam are the same as the usually used in Openfoam. Meaning that we have:
[mass(kg),length(m),time(s),temperature(K),quantity(mol),current(A),luminous intensity (cd)].
Because I see in the tutorial for the molar mass of H+:
Mw Mw [1 0 0 0 -1 0 0] 1;
Which would imply that either the quantity is in kmol or the mass is in g. Could you tell me if this due to an error in the tutorials or just to the way the units are defined in Geochemfoam.

Thanks again for the help.
Sincerely,
Santhosh

**Comment by julienmaes** [comment]
Hi Santosh,

Same unit, except for quantity for which it's kmol and not mol. In my opinion it is a mistake for openfoam to use mol, the SI is kmol (of course, since it's kg and not g...)

**Comment by Santhosh91000** [comment]
I see that makes sense. Thanks a lot for the quick reply! 

Have a nice end of day,
Sincerely,
Santhosh
